CMakeLists.txt

```diff
@@ -125,6 +129,31 @@ elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
 	include_directories(SYSTEM "include/support/osx")
 endif()
 
+# If we're on windows, do a check for what compiler we're using (so we can link the appropriate libraries)
+if(WIN32)
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") # We're dealing with MinGW etc
+        set(Boost_USE_STATIC_LIBS ON)
+        set(BOOST_INCLUDEDIR ${LIB_ROOT_DIR}/boost/boost)
+
+        # Do a check for system bitness
+        if(AMD64)
+            set(BOOST_LIBRARYDIR ${LIB_ROOT_DIR}/boost/lib/amd64/debug)
+        else()
+            set(BOOST_LIBRARYDIR ${LIB_ROOT_DIR}/boost/lib/x86/debug)
+        endif()
+
+        set(GLEW_LIBRARY ${LIB_ROOT_DIR}/GLEW/lib/libglew.a)
+        set(SDL2_LIBRARY ${LIB_ROOT_DIR}/SDL2/lib)
+        set(ZLIB_LIBRARY ${LIB_ROOT_DIR}/zlib/lib/zlib.a)
+
+        set(GLEW_LIBRARIES ${LIB_ROOT_DIR}/GLEW/lib/libglew.a)
+        set(SDL2_INCLUDE_DIRS ${LIB_ROOT_DIR}/SDL2/include/SDL2) # CMake doesn't automatically pass this like the others???
+        set(CMAKE_EXE_LINKER_FLAGS "-lmingw32")
```

Comment:
Does GCC on Windows imply MinGW? What about Cygwin?

Reply:
I assume so, but I'll need to double check. Considering they're both built off of GCC.